torchdynamo/utils.py

```diff
@@ -37,6 +37,44 @@
 
 log = logging.getLogger(__name__)
 
+# profiling compilation time
+compilation_metrics = collections.OrderedDict()
+@contextlib.contextmanager
+def timed_compile(key):
+    global compilation_metrics
+    pop = compilation_metrics
+    assert key not in compilation_metrics, "trying to double-record"
```

Comment:
What if we recursively re-enter? This feels like it imposes runtime ordering asserts. Why do we need to do this pop and store dance? I'm a little too dumb to understand this logic.  Can we:

1) make t0
2) call the func
3) make t1
4) store t1 in a `dict[name, list[time]]`